contracts/core/EntryPoint.sol

```diff
@@ -232,32 +233,25 @@ contract EntryPoint is IEntryPoint, StakeManager {
      *      In order to split the running opcodes of the wallet (validateUserOp) from the paymaster's validatePaymasterUserOp,
      *      it should look for the NUMBER opcode at depth=1 (which itself is a banned opcode)
      * @param userOp the user operation to validate.
-     * @param offChainSigCheck if the wallet has an aggregator, skip on-chain aggregation check. In thus case, the bundler must
-     *          perform the equivalent check using an off-chain library code
      * @return preOpGas total gas used by validation (including contract creation)
      * @return prefund the amount the wallet had to prefund (zero in case a paymaster pays)
      * @return deadline until what time this userOp is valid (the minimum value of wallet and paymaster's deadline)
-     * @return actualAggregator the aggregator used by this userOp. if a non-zero aggregator is returned, the bundler must get its params using
-     *      aggregator.
-     * @return sigForUserOp - only if has actualAggregator: this value is returned from IAggregator.validateUserOpSignature, and should be placed in the userOp.signature when creating a bundle.
-     * @return sigForAggregation  - only if has actualAggregator:  this value is returned from IAggregator.validateUserOpSignature, and should be passed to aggregator.aggregateSignatures
-     * @return offChainSigInfo - if has actualAggregator, and offChainSigCheck is true, this value should be used by the off-chain signature code (e.g. it contains the sender's publickey)
      */
-    function simulateValidation(UserOperation calldata userOp, bool offChainSigCheck)
-    external returns (uint256 preOpGas, uint256 prefund, uint256 deadline, address actualAggregator, bytes memory sigForUserOp, bytes memory sigForAggregation, bytes memory offChainSigInfo) {
+    function simulateValidation(UserOperation calldata userOp)
+    external returns (uint256 preOpGas, uint256 prefund, uint256 deadline) {
         uint256 preGas = gasleft();
 
         UserOpInfo memory outOpInfo;
 
-        (actualAggregator, deadline) = _validatePrepayment(0, userOp, outOpInfo, SIMULATE_NO_AGGREGATOR);
+        address aggregator;
+        (aggregator, deadline) = _validatePrepayment(0, userOp, outOpInfo, SIMULATE_FIND_AGGREGATOR);
         prefund = outOpInfo.prefund;
         preOpGas = preGas - gasleft() + userOp.preVerificationGas;
 
-        numberMarker();
-        if (actualAggregator != address(0)) {
-            (sigForUserOp, sigForAggregation, offChainSigInfo) = IAggregator(actualAggregator).validateUserOpSignature(userOp, offChainSigCheck);
+        if (aggregator != address(0)) {
+            revert UnverifiedSignatureAggregator(preOpGas, prefund, deadline, aggregator);
         }
-        require(msg.sender == address(0), "must be called off-chain with from=zero-addr");
+        require(tx.origin == address(0), "must be called off-chain with from=zero-addr");
```

Comment:
I think this line should be in the first line as it checks the pre-condition that the method is called from a right place and as a view call only.

Reply:
The line is last, so that we can see other revert reasons.
in solidity, there is no side-effect for running reverts at the end (except gas usage, which we don't care about in view function)
anyway, it was removed, as we now always revert, for simplified API.